autoload/ale/util.vim

```diff
@@ -90,6 +90,17 @@ function! ale#util#LocItemCompare(left, right) abort
         return 1
     endif
 
+    " put warnings after errors (for the same line) since the text that shows
+    " when the cursor is moved will show only the first entry
+
+    if a:left['type'] < a:right['type']
+      return -1
+    endif
+
+    if a:left['type'] > a:right['type']
+      return 1
+    endif
```

Comment:
This change will sort the items in the wrong order, which is important for the movement commands, etc. Instead of getting items in column order `1, 2, 3, 4, 5`, you can see items in order `E3, E4, W1, W2, W5`, etc.

You should move these lines down below the column sorting, so it's the third key items are sorted on. Then `E` will take priority over `W` if there are both warnings and errors for the same column, if that's the problem. The way the items are sorted, the cursor is supposed to show information for a problem near the cursor. It doesn't matter what the severity is.